crates/wasi-demo-app/build.rs

```diff
@@ -0,0 +1,62 @@
+#[cfg(feature = "oci-v1-tar")]
+use {
+    anyhow::Context, oci_spec::image as spec, oci_tar_builder::Builder, sha256::try_digest,
+    std::env, std::fs::File, std::path::PathBuf,
+};
+
+#[cfg(not(feature = "oci-v1-tar"))]
+fn main() {}
+
+#[cfg(feature = "oci-v1-tar")]
+fn main() {
+    env_logger::init();
+
+    let out_dir = PathBuf::from(env::var("OUT_DIR").unwrap());
+    let p = out_dir.join("img.tar");
+    let bin_output_dir = out_dir
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap();
+
+    let app_path = bin_output_dir.join("wasi-demo-app.wasm");
+    let layer_path = out_dir.join("layer.tar");
+    tar::Builder::new(File::create(&layer_path).unwrap())
+        .append_path_with_name(&app_path, "wasi-demo-app.wasm")
+        .unwrap();
+
+    let mut builder = Builder::default();
+
+    builder.add_layer(&layer_path);
+
+    let config = spec::ConfigBuilder::default()
+        .entrypoint(vec!["/wasi-demo-app.wasm".to_owned()])
+        .build()
+        .unwrap();
+
+    let layer_digest = try_digest(layer_path.as_path()).unwrap();
+    let img = spec::ImageConfigurationBuilder::default()
+        .config(config)
+        .os("wasi")
+        .architecture("wasm32")
```

Comment:
OCI prefers `wasm` instead of `wasm32`, see the [discussion in this PR](https://github.com/opencontainers/image-spec/pull/964#issuecomment-1289408465)

Reply:
When I tried this out I had to run `rustup target add wasm32-wasi`. I guess this is related?

Reply:
I don't think it's related. The architecture here is for the OCI image, not for cargo or the rust code

Reply:
i see that now, thanks!

Reply:
Fixed